source/client_channel_handler.c

```diff
@@ -799,6 +842,14 @@ static void s_request_outgoing_task(struct aws_channel_task *task, void *arg, en
                 aws_mqtt_connection_statistics_change_operation_statistic_state(
                     request->connection, request, AWS_MQTT_OSS_NONE);
 
+                /**
+                 * Cache the time of the write on the socket so long as the packet was not a ping packet.
+                 * We do this to avoid the PING packet overriding the ping interval
+                 */
+                if (request->packet_type != AWS_MQTT_PACKET_PINGREQ) {
+                    aws_high_res_clock_get_ticks(&connection->last_outbound_socket_write_time);
+                }
+
```

Comment:
*Note* We can avoid needing to pass the packet type entirely by just checking the packet size and seeing if it is over `0`, as we assign a packet size to everything but the PING tasks currently.

However, I was not totally sure if this was the right approach because there is a possibility (though likely slim) that we have packets going out that we do not necessarily want to track the size of, but we do want to have reset the ping. Examples of this could be QoS2 ACK packets.

Either approach should work though, but I thought it would be better to be more targeted in the condition checking for the PING packet than not.

Reply:
If we remove adding the type along with the request, then we can trim the changes this PR makes without any loss of functionality currently.

Reply:
I have changed my mind again. For now, we can just use the operation size and should we need to change it in the future, we can do so then. No reason to prematurely optimize and add quite a bit of code for a *maybe*.